core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -164,17 +165,35 @@ class ReplicaAlterLogDirsThread(name: String,
   def fetchEpochsFromLeader(partitions: Map[TopicPartition, Int]): Map[TopicPartition, EpochEndOffset] = {
     partitions.map { case (tp, epoch) =>
       try {
-        tp -> new EpochEndOffset(Errors.NONE, replicaMgr.getReplicaOrException(tp).epochs.get.endOffsetFor(epoch))
+        val (leaderEpoch, leaderOffset) = replicaMgr.getReplicaOrException(tp).epochs.get.endOffsetFor(epoch)
+        val leaderEpochInResponse: Int =
+          if (brokerConfig.interBrokerProtocolVersion >= KAFKA_2_0_IV0) leaderEpoch
```

Comment:
Do we need this check? Since we are getting the leader epoch from the current replica's log directly, even when IBP < KAFKA_2_0_IV0, it seems that we can just return leaderEpoch.

Reply:
If we are on protocol < 2.0, then the local replica will be fetching from leader based on older protocol (not using leader epoch).  If we don't check here, the future replica will be fetching from the local replica based on leader epoch. Seems inconsistent? On the other hand, it should still work for the future replica to truncate using leader epoch in that  case too.

Reply:
If we keep the leader epoch here for better outcome, should we still check `useLeaderEpochInResponse` in getOffsetTruncationState() so that it returns `OffsetTruncationState(min(leaderEpochOffset.endOffset, replica.logEndOffset.messageOffset), truncationCompleted = true)` if `useLeaderEpochInResponse` is false?

Reply:
If we use leader epoch, then we should go all the way using the new protocol, i.e., continue truncating until finding the consistent point. 
Ok, I will change back to using leader epoch if available for future replica.